### ISSUE TITLE
Understanding `test/Rings/nmod-localizations.jl` and `test/Rings/integer-localizations.jl`

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/Oscar.jl#2118]
I just came upon `test/Rings/nmod-localizations.jl` (CC @HechtiDerLachs) and was surprised by this test file, as it ...
- contains `export` statements (!)
- defines three new `mutable struct` types (`NmodComplementOfPrimeIdeal`, `NmodLocalizedRing`, `NmodLocalizedRingElem`)
- defines lots of functions and methods, several even with full docstrings
- and then 20 out of 168 lines test the code defined above it.

This seems odd... Perhaps this file should really live inside `experimental`? Or what is the plan with this code, if any?

**Comment by fingolfin** [comment]
Ah, something similar applies to `test/Rings/integer-localizations.jl`

**Comment by HechtiDerLachs** [comment]
Yes, this was in experimental once. But then people said "Please remove it, because it's interfering with our implementations elsewhere." We eventually agreed to move it to the tests so that people also have a nice sample for how to implement the localization interface. Concerning the export statements: I probably was not aware of such things at the time. But it's no rare thing that I define new structs in the tests, in particular to test interfaces. Anything wrong with that?

**Comment by fingolfin** [comment]
I see, thanks for the background.

If these are examples, then they should be in `examples`, not `test`.

**Comment by thofma** [comment]
They are tests for the interface. I think those should be tests. Maybe not all of them?
